go/database/mpt/live_trie.go

```diff
@@ -158,6 +159,32 @@ func (s *LiveTrie) VisitTrie(visitor NodeVisitor) error {
 	return s.forest.VisitTrie(&s.root, visitor)
 }
 
+// VisitAccount visits the storage nodes of an account with the given address.
+// The visited nodes contain the account node and all storage nodes of the account.
+func (s *LiveTrie) VisitAccount(address common.Address, visitor NodeVisitor) error {
+	var innerError error
+	// this visitor finds the account node with the given address
+	accountVisitor := MakeVisitor(func(node Node, info NodeInfo) VisitResponse {
+		switch n := node.(type) {
+		case *AccountNode:
+			if n.Address() == address {
+				// and then it sends the storage node to the visitor
+				_, innerError = n.Visit(s.forest, &n.storage, 0, visitor)
```

Comment:
Is it important that the `AccountNode` is visited first before visiting the storage or is this irrelevant for the intended use case?

Reply:
not sure I understand this comment, I have to find the account first, before I can start to iterate its storage, right?

Reply:
The visitor getting passed to the `VisitAccountStorage` function will be called first with the account node, followed by all the branch, extension, and value nodes of the account's state. My question was whether this initial call with the account node is essential or whether just visiting the trie forming the Storage state is sufficient.

If the `AccountNode` needs to be visited for your use case, the function should probably be renamed to `VisitAccountAndStorage`. If only the storage is relevant, the documentation should be updated and the visit process should be started with the root node of the storage instead of with the `AccountNode` referencing the root.

Reply:
oh, got it. No I did not need the account node in this use case, it was just the most straightforward to implement. Changed not to return the account node.